client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -78,13 +90,27 @@ public T next() {
                                        }
                                        return buffer;
                                    } catch (NoSuchSegmentException | SegmentTruncatedException e) {
+                                       handleSegmentTruncated(segment);
                                        throw new TruncatedDataException("Segment " + segment + " has been truncated.");
```

Comment:
It throws a `TruncatedDataException` in the impl class, but there is no hint for this from the base method `next()`. Can we improve it?

Reply:
Added java docs https://github.com/pravega/pravega/blob/7bf8054c45d5cf3755a09724aa34b87a15ccb702/client/src/main/java/io/pravega/client/batch/SegmentIterator.java#L31C1-L31C1